runhouse/servers/cluster_servlet.py

```diff
@@ -486,10 +549,16 @@ async def astatus(self, send_to_den: bool = False) -> Tuple[Dict, Optional[int]]
             else None
         )
 
+        # rest the gpu_info only after the status was sent to den. If we should not send status to den,
+        # self.gpu_info will not be updated at all, therefore should not be reset.
+        if send_to_den:
```

Comment:
move this into a helper for reuse? maybe `reset_gpu_info`?

Reply:
actually not sure it can be reused,  because EnvSerlvet and the ClusterServlet have different `self.lock` and `self.gpu_info`. 

Moreover, the self.gpu_info is reseted only once in the EnvSerlvet and once in the ClusterServlet.